packages/amplify-provider-awscloudformation/src/system-config-manager.ts

```diff
@@ -297,10 +297,10 @@ export function getProfileCredentials(profileName: string) {
 
 function validateCredentials(credentials: $TSAny, profileName: string) {
   const missingKeys = [];
-  if (!credentials?.accessKeyId) {
+  if (!credentials?.accessKeyId && !process.env.AWS_ACCESS_KEY_ID) {
     missingKeys.push('aws_access_key_id');
   }
-  if (!credentials?.secretAccessKey) {
+  if (!credentials?.secretAccessKey && !process.env.AWS_SECRET_ACCESS_KEY) {
```

Comment:
I was under the impression that this function is only be called for validating profile creds. This logic should be skipped altogether if using env vars iirc

See line 89

Reply:
The problem is that we have a customer who wants to use a profile which authenticates using the env var override, if we want to not support that flow we can close this PR and mark the issue as won't fix

Reply:
My thought is that it might be a CICD thing where they're using contextual creds, say from EC2, and want to have a version controlled .aws/config file. It's a pretty specific use case but it's not invalid, I think it might make sense to allow something like this flow through some kind of escape hatch where we say hey do whatever you want and if it works through the AWS CLI, it's fine, but we won't give protections or validations.

I can understand not wanting to make the credential validation too complex because of all the permutations, but it's also probably not great for us to block alternative flows entirely. What's your feeling on it? Close the PR as a won't do, new escape hatch, or get some more feedback